src/node/rpc/member_frontend.h

```diff
@@ -1332,16 +1332,61 @@ namespace ccf
         }
 
         const auto in = params.get<Propose::In>();
-        const auto proposal_id =
-          fmt::format("{:02x}", fmt::join(caller_identity.request_digest, ""));
 
-        Proposal proposal(in.script, in.parameter, caller_identity.member_id);
+        if (!consensus)
+        {
+          return make_error(
+            HTTP_STATUS_INTERNAL_SERVER_ERROR,
+            ccf::errors::InternalError,
+            "No consensus available.");
+        }
+
+        std::string proposal_id;
 
+        if (consensus->type() == ConsensusType::CFT)
+        {
+          auto root_at_read = ctx.tx.get_root_at_read_version();
```

Comment:
I thought this was supposed to take a read dependency on the signatures table to avoid the compaction issue discussed a couple of weeks ago?

Reply:
That may be an implementation detail in #2136 for BFT, but unnecessary for CFT, since it's not a historical root.